### PR TITLE
Update to align to genesearch API, add consistent API error handling

### PR DESCRIPTION
**Pull request by dauglyon**
- Fixes the comma-joining bug which was causing searches to fail silently (a fix for the "silent" failure is up on the genesearch BE)
- Fix existing tests and add new tests for FASTA parsing
- Fixes the E-Value field so that it actually functions!
- Adds better error handling for API calls, and now surfaces to the UI all unhandled API request/parsing errors by default.
- Add "failed" search state which was introduced to avoid silent search failures on the BE
- Removed usePrevious.tsx (unused code)

### CODE REVIEW

File: react_app/src/components/SearchResults.tsx [hunk 1]
```diff
@@ -77,8 +77,11 @@ export const SearchResults: FC<SearchResultsProps> = ({ searches, addSearchById,
               <br />
               <List.Item.Meta
                 avatar={(
-                  <Spin spinning={search.status !== "completed"}>
-                    <Avatar>{search.status === "completed" ? '\u2714' : ''}</Avatar>
+                  <Spin spinning={!["completed", "failed"].includes(search.status ?? "")}>
+                    <Avatar>
+                      {search.status === "completed" ? '\u2713' : ''}
+                      {search.status === "failed" ? '\u2715' : ''}
```
Comment (dauglyon) [inline react_app/src/components/SearchResults.tsx:83]: Added UI state for failed searches

File: react_app/src/hooks/useNamespaces.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { useEffect, useState } from 'react';
 import { Namespace } from '../types/Namespace';
+import { handleApiCall } from '../utils/handleApiCall';
```
Comment (dauglyon) [inline react_app/src/hooks/useNamespaces.tsx:3]: Updated namespace API call to use `handleApiCall`

File: react_app/src/hooks/useSearches.tsx [hunk 3]
```diff
@@ -1,52 +1,30 @@
 import { useEffect, useState } from 'react';
-import { notification } from 'antd';
 import { useLocalStorage } from '@rehooks/local-storage';
 import { Search } from '../types/Search';
 import { Namespace } from '../types/Namespace';
 import { urlFor } from '../utils/urlFor';
 import { parseFASTA } from '../utils/handleFASTA';
+import { handleApiCall } from '../utils/handleApiCall';
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:7]: Updated all search API calls to use `handleApiCall`

File: react_app/src/utils/handleApiCall.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { notification } from "antd";
+
+interface HandleApiCallParams<T, R> {
+  call: () => ReturnType<typeof fetch>;
+  handleResult: (json: T) => R;
+  errContext: string;
+  handleError?: (error: Error) => boolean | void | Promise<boolean | void>;
+}
+
```
Comment (dauglyon) [inline react_app/src/utils/handleApiCall.ts:9]: This function is a helper to wrap & catch errors in both calls to the API and the parsing of the response. Once there is a stable version of the backend, the `handleResult` parsing callback could be extracted in favor of passing a typegaurd

File: react_app/src/utils/handleFASTA.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export async function parseFASTA(fasta: string) {
   const sequences = sequenceGroups.map(([header, start, stop]) => {
     const idMatch = lines[header].match(/^>\s?([^ ]+)/);
     const id = idMatch ? idMatch[1] : `Unknown ID (L${header})`;
-    const seq = lines.slice(start, stop).join();
+    const seq = lines.slice(start, stop).join("");
```
Comment (dauglyon) [inline react_app/src/utils/handleFASTA.ts:42]: fixes bug where sequences were sent to the server joined by a comma

File: react_app/src/utils/handleFASTA.test.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
-import { validateFASTA } from "./handleFASTA";
+import { validateFASTA, parseFASTA } from "./handleFASTA";
```
Comment (dauglyon) [inline react_app/src/utils/handleFASTA.test.ts:1]: Added tests for `parseFASTA` and fixed `validateFASTA` tests so as not to have unhandled promises

File: react_app/src/components/SearchForm.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import React, { FC } from 'react';
-import { Spin, Form, Input, Select, InputNumber, Button, PageHeader, Alert } from 'antd';
+import React, { FC, useState } from 'react';
```
Comment (dauglyon) [inline react_app/src/components/SearchForm.tsx]: In this file: Changed search form E-Value input to exponential notation, changed error title on namespace load failure

**Review by dakotablair — APPROVED** [review]
These changes look good to me!

**Pull request merged**